src/saturn_engine/worker/executors/bootstrap.py

```diff
@@ -25,11 +31,53 @@ def __init__(self, initialized_hook: EventHook["PipelineBootstrap"]):
 
         self.logger = logging.getLogger("saturn.bootstrap")
 
+    def _process_error(
+        self,
+        outputs: dict[str, list[TopicItem]],
+        exc_value: BaseException,
+        exc_traceback: TracebackType,
+    ) -> list[PipelineOutput]:
+        # Match the exception and consume it if found, else reraise the exception
+        for channel in outputs.keys():
+            # TODO: better way to identify error output, maybe seperate field instead
+            if is_output_error_handler(channel):
+                exc_filter = parse_warning_filter(channel, escape=False)
+                exc_details = get_exception_details(exc_value, exc_traceback)
+
+                if does_error_match(exception=exc_details, exception_filter=exc_filter):
+                    return [
+                        PipelineOutput(
+                            channel=channel,
+                            message=TopicMessage(
+                                args=asdict(
+                                    ErrorMessageArgs(
+                                        type=exc_details.exception_type.__name__,
+                                        module=exc_details.exception_type.__module__,
+                                        message=str(exc_value),
+                                        # traceback=TracebackData.from_exception(exc_value),
+                                    )
+                                )
+                            ),
+                        )
+                    ]
+        # Unhandled exception
+        raise exc_value
+
     def bootstrap_pipeline(self, message: PipelineMessage) -> PipelineResults:
         return self.pipeline_hook.emit(self.run_pipeline)(message)
 
     def run_pipeline(self, message: PipelineMessage) -> PipelineResults:
-        execute_result = message.execute()
+        try:
+            execute_result = message.execute()
+        except Exception as err:
+            _, exc_value, exc_traceback = sys.exc_info()
+            # Shouldn't be None but we want to make mypy happy
+            if exc_value is None or exc_traceback is None:
+                self.logger.exception("Failed to catch context for exception")
+                raise err
+            execute_result = self._process_error(
+                message.outputs, exc_value, exc_traceback
```

Comment:
`outputs` is a pipeline property, not a message property and the executor should not be aware of this concept. Executor only job is to run pipeline and return results, let's not try to add more work to it.

Output are to be handled in the `ExecutorQueue`.